src/sysupdate/sysupdate.c

```diff
@@ -1862,11 +1999,11 @@ static int parse_argv(int argc, char *argv[]) {
         if (arg_definitions && arg_component)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "The --definitions= and --component= switches may not be combined.");
 
-        if (arg_definitions && arg_next)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "The --definitions= and --next switches may not be combined.");
+        if (arg_definitions && arg_stream)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "The --definitions= and --stream= switches may not be combined.");
 
-        if (arg_component && arg_next)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "The --component= and --next switches may not be combined.");
+        if (arg_component && arg_stream)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "The --component= and --stream= switches may not be combined.");
```

Comment:
Do we really need this restriction now we're using `--stream`?  I think it'd be useful (and simpler) to allow any component to define streams and allow the combination of options.  The only weirdness would be if a user tried to update all with `--stream` passed, but we could either make this use the default target or _require_ `--component` with `--stream`.

Reply:
> The only weirdness would be if a user tried to update all with --stream passed

`updatectl` shouldn't be exposing `--stream` on the command line like this. Instead the different available streams should become their own targets. i.e. `host` is the host system (`/usr/lib/syuspdate.d`) and `stream:foobar` is the foobar stream of the host system (`--stream=foobar`, `/usr/lib/sysupdate@foobar.d`).

Stream targets should be impossible to update via `updatectl update`, to avoid situations where the user might try to `updatectl update host stream:foobar` and break their system (since both of those targets update the same partitions on disk!). Consequently, stream targets should not be enumerated by default for things like `updatectl update` (with no args), but probably can be checked by commands like `updatectl check`

> require --component with --stream

Host system is the most important  user of streams, and if we require `--component` with it the host won't be allowed to use it

> Do we really need this restriction now we're using `--stream`

I'd prefer to keep it for the time being, because bringing components into the mix makes life difficult.

Namely: With the host system we can easily track the "current" stream because switching streams swaps out the contents of /usr. So then by definition whatever is in /usr/lib/sysupdate.d is the "current" stream and all instances of `/usr/lib/sysupdate@<name>.d` are alternative streams we can switch to. With components the same does not apply: components by definition have no influence over the contents of /usr, and so there is no implicit "current" stream - we'd need some way to track which stream is current for a given component. I don't have an idea about how to neatly pull that off.

If it ever becomes necessary to have update streams for components, we can work on solving the problem and can lift the restriction. But for now let's not do that. I can't think something that belongs as its own component and yet has a complicated enough release cycle to have more than one stream; there's no such thing as "shim nightly" or whatever.